src/anim/controller/anim-blend-tree.js

```diff
@@ -30,19 +12,19 @@ function clamp(num, min, max) {
  * @param {AnimBlendTree|null} parent - The parent of the AnimBlendTree. If not null, the AnimNode is stored as part of a {@link AnimBlendTree} hierarchy.
  * @param {string} name - The name of the BlendTree. Used when assigning a {@link AnimTrack} to its children.
  * @param {number|Vec2} point - The coordinate/vector thats used to determine the weight of this node when it's part of a {@link AnimBlendTree}.
- * @param {string} type - Determines which blending algorithm is used to calculate the weights of its child nodes. One of ANIM_BLEND_*.
  * @param {string[]} parameters - The anim component parameters which are used to calculate the current weights of the blend trees children.
  * @param {object[]} children - The child nodes that this blend tree should create. Can either be of type {@link AnimNode} or {@link BlendTree}.
  * @param {Function} findParameter - Used at runtime to get the current parameter values.
  */
 class AnimBlendTree extends AnimNode {
-    constructor(state, parent, name, point, type, parameters, children, findParameter) {
+    constructor(state, parent, name, point, parameters, children, findParameter) {
         super(state, parent, name, point);
-        this._type = type;
         this._parameters = parameters;
         this._parameterValues = null;
         this._children = [];
         this._findParameter = findParameter;
+        this._p = new Vec2();
+        this._pip = new Vec2();
```

Comment:
if these are optimizations to avoid allocations, perhaps they should be static, or global.

Reply:
Sure, i'll set them as static properties.

Reply:
Con for static: they'll be visible/accessible to devs poking at our structures. Typically we declare these types of vars at global module scope.

https://github.com/playcanvas/engine/blob/master/src/scene/graph-node.js#L9-L23

But I'm not _too_ precious about it.